service/src/test/java/greencity/service/AchievementServiceImplTest.java

```diff
@@ -42,4 +60,120 @@ void findAllWithOneValueInRepoTest() {
         List<AchievementDTO> findAllResult = achievementService.findAll();
         assertEquals(1L, (long) findAllResult.get(0).getId());
     }
+
+    @Test
+    void findAllByPageableTest() {
+        Pageable pageable = PageRequest.of(0,2);
+        Achievement achievement = new Achievement();
+        AchievementVO achievementVO = new AchievementVO();
+        Page<Achievement> pages = new PageImpl<>(Collections.singletonList(achievement),pageable,10);
+        when(achievementRepo.findAll(pageable)).thenReturn(pages);
+        when(modelMapper.map(achievement, AchievementVO.class)).thenReturn(achievementVO);
+        PageableAdvancedDto<AchievementVO> pageableAdvancedDto = achievementService.findAll(pageable);
+        assertEquals(10, pageableAdvancedDto.getTotalElements());
+    }
+
+    @Test
+    void searchAchievementByTest() {
+        Pageable pageable = PageRequest.of(0,2);
+        Achievement achievement = new Achievement();
+        Page<Achievement> page = new PageImpl<>(Collections.singletonList(achievement),pageable,10);
+        AchievementVO achievementVO = new AchievementVO();
+        when(achievementRepo.searchAchievementsBy(pageable,"")).thenReturn(page);
+        when(modelMapper.map(achievement, AchievementVO.class)).thenReturn(achievementVO);
+        PageableAdvancedDto<AchievementVO> pageableAdvancedDto = achievementService.searchAchievementBy(pageable,"");
+        assertEquals(10, pageableAdvancedDto.getTotalElements());
+    }
+
+    @Test
+    public void saveTest() {
+        Achievement achievement = new Achievement();
+        achievement.setTranslations(new ArrayList<>());
+        AchievementCategory achievementCategory = new AchievementCategory();
+        AchievementCategoryDto achievementCategoryDto = new AchievementCategoryDto("Test");
+        AchievementPostDto achievementPostDto = new AchievementPostDto();
+        achievementPostDto.setAchievementCategory(achievementCategoryDto);
+        AchievementCategoryVO achievementCategoryVO = new AchievementCategoryVO();
+        AchievementVO achievementVO = new AchievementVO();
+        UserVO userVO = new UserVO();
+        UserAchievementVO userAchievement = new UserAchievementVO();
```

Comment:
Better to write getters in ModelUtils for creating test entities objects.

Reply:
Done